src/neo/SmartContract/Native/Designate/DesignateContract.cs

```diff
@@ -37,6 +37,37 @@ public ECPoint[] GetDesignatedByRole(StoreView snapshot, Role role, uint index)
                 .FirstOrDefault() ?? System.Array.Empty<ECPoint>();
         }
 
+        [ContractMethod(0_01000000, CallFlags.AllowModifyStates)]
+        private byte[] GetDesignatedInfo(StoreView snapshot, Role role, ECPoint node)
+        {
+            var nodes = GetDesignatedByRole(snapshot, role, snapshot.HeaderHeight);
+            if (!nodes.Contains(node))
+                throw new InvalidOperationException("Node not found");
+
+            var value = snapshot.Storages.TryGet(CreateStorageKey((byte)role).Add(node));
+            return value?.Value ?? System.Array.Empty<byte>();
+        }
+
+        [ContractMethod(0_05000000, CallFlags.AllowModifyStates)]
+        private void SetDesignatedInfo(ApplicationEngine engine, Role role, ECPoint node, byte[] value)
```

Comment:
The `value` is encrypted, right?

Reply:
raw, we can choose of-chain

Reply:
I think exposing IP addresses on the chain is very dangerous.

Reply:
They can use a shared key

Reply:
> I think exposing IP addresses on the chain is very dangerous.

This should not be done. We should not share IP and identify clients. This is an assumption of Neo's topology.

Reply:
But what is related to IP here?

Reply:
Oracle nodes, state validators, neofs validators can set their encrypted ip addresses onchain. Then they can communicate with each other directly by rpc endpoint.

Reply:
It does not sounds good to me.

Reply:
But we need some way to share IP addresses.

Reply:
This changes the topology of the network and open a door for attacks if any of these nodes get hacked.

Reply:
However, we still have other ways to get the IP addresses of the oracle node, such as making a request to our own service. 

So, I think this is an acceptable method.

Reply:
> such as making a request to our own service.

I think we need a decentralized way.

Reply:
Do it in stages? Currently, allow oracle nodes to access them directly. The next step to design an anonymous access protocol.